src/test/ui/xcrate/xcrate-private-by-default.stderr

```diff
@@ -2,61 +2,151 @@ error[E0603]: static `j` is private
+   |
+note: enum `m` is declared here
+  --> $DIR/auxiliary/static_priv_by_default.rs:50:1
+   |
+LL | enum m {}
+   | ^^^^^^^^^
 
 error[E0603]: type alias `n` is private
   --> $DIR/xcrate-private-by-default.rs:31:35
    |
 LL |     foo::<static_priv_by_default::n>();
-   |                                   ^
+   |                                   ^ this type alias is private
+   |
+note: type alias `n` is declared here
+  --> $DIR/auxiliary/static_priv_by_default.rs:51:1
+   |
+LL | type n = isize;
+   | ^^^^^^^^^^^^^^^
 
 error[E0603]: module `foo` is private
   --> $DIR/xcrate-private-by-default.rs:35:29
    |
 LL |     static_priv_by_default::foo::a;
-   |                             ^^^
+   |                             ^^^ this module is private
+   |
+note: module `foo` is declared here
+  --> $DIR/auxiliary/static_priv_by_default.rs:12:1
+   |
+LL | / mod foo {
+LL | |     pub static a: isize = 0;
+LL | |     pub fn b() {}
+LL | |     pub struct c;
+...  |
+LL | |     pub type reexported_e = isize;
+LL | | }
+   | |_^
 
 error[E0603]: module `foo` is private
   --> $DIR/xcrate-private-by-default.rs:37:29
    |
 LL |     static_priv_by_default::foo::b;
-   |                             ^^^
+   |                             ^^^ this module is private
+   |
+note: module `foo` is declared here
+  --> $DIR/auxiliary/static_priv_by_default.rs:12:1
+   |
+LL | / mod foo {
+LL | |     pub static a: isize = 0;
+LL | |     pub fn b() {}
+LL | |     pub struct c;
+...  |
+LL | |     pub type reexported_e = isize;
+LL | | }
+   | |_^
 
 error[E0603]: module `foo` is private
   --> $DIR/xcrate-private-by-default.rs:39:29
    |
 LL |     static_priv_by_default::foo::c;
-   |                             ^^^
+   |                             ^^^ this module is private
+   |
+note: module `foo` is declared here
+  --> $DIR/auxiliary/static_priv_by_default.rs:12:1
+   |
+LL | / mod foo {
+LL | |     pub static a: isize = 0;
+LL | |     pub fn b() {}
+LL | |     pub struct c;
+...  |
+LL | |     pub type reexported_e = isize;
+LL | | }
+   | |_^
 
 error[E0603]: module `foo` is private
   --> $DIR/xcrate-private-by-default.rs:41:35
    |
 LL |     foo::<static_priv_by_default::foo::d>();
-   |                                   ^^^
+   |                                   ^^^ this module is private
+   |
+note: module `foo` is declared here
+  --> $DIR/auxiliary/static_priv_by_default.rs:12:1
+   |
+LL | / mod foo {
+LL | |     pub static a: isize = 0;
+LL | |     pub fn b() {}
+LL | |     pub struct c;
+...  |
+LL | |     pub type reexported_e = isize;
+LL | | }
+   | |_^
 
 error[E0603]: module `foo` is private
   --> $DIR/xcrate-private-by-default.rs:43:35
    |
 LL |     foo::<static_priv_by_default::foo::e>();
-   |                                   ^^^
+   |                                   ^^^ this module is private
+   |
+note: module `foo` is declared here
+  --> $DIR/auxiliary/static_priv_by_default.rs:12:1
+   |
+LL | / mod foo {
```

Comment:
It would be better to use the item header span (`def_span`, IIRC) instead of the full item span here.